src/resource_reader.rs

```diff
@@ -13,14 +17,39 @@ pub struct FileRead {
 
 pub struct ResourceReader {
     runtime: TargetRuntime,
+    // Cache file content, path->content
+    cache: Arc<Mutex<HashMap<String, String>>>,
```

Comment:
Why use an async-lock? I think a simple sync::Mutex should work for our usecase.

Reply:
Performance is not a major concern in this part of the application.